eZ/Publish/Core/Limitation/ObjectStateLimitationType.php

```diff
@@ -236,8 +236,49 @@ public function getCriterion(APILimitationValue $value, APIUserReference $curren
             return new Criterion\ObjectStateId($value->limitationValues[0]);
         }
 
-        // several limitation values: IN operation
-        return new Criterion\ObjectStateId($value->limitationValues);
+        $groupedLimitationValues = $this->groupLimitationValues($value->limitationValues);
+
+        if (count($groupedLimitationValues) === 1) {
+            // several limitation values: IN operation
+            return new Criterion\ObjectStateId($value->limitationValues);
```

Comment:
The check that you proposed is done in another form here: https://github.com/natanael89/ezpublish-kernel/blob/a5af4ac2731f3f5e8dee28e8a6a1bc9e2fae8b3b/eZ/Publish/Core/Limitation/ObjectStateLimitationType.php#L234. It checks if there is only one limitation value and it indeed is done before the call to `groupLimitationValues`. However, there can also be the case when there are multiple limitation values, but all of them are coming from the same state group. This case is what ` if (count($groupedLimitationValues) === 1)` is checking. It requires grouping them first though.